lib/lz4.c

```diff
@@ -886,8 +886,8 @@ LZ4_FORCE_INLINE int LZ4_compress_generic_validated(
     /* the dictCtx currentOffset is indexed on the start of the dictionary,
      * while a dictionary in the current context precedes the currentOffset */
     const BYTE* dictBase = !dictionary ? NULL : (dictDirective == usingDictCtx) ?
-                            dictionary + dictSize - dictCtx->currentOffset :
-                            dictionary + dictSize - startIndex;
+                            (const BYTE*) ((uptrval) dictionary + dictSize - dictCtx->currentOffset) :
```

Comment:
This one is strange.
We just checked `!dictionary` just before,
so by that point, we are certain than `dictionary != NULL`,
hence it shouldn't be necessary to fudge the calculation through `uptrval`.

Reply:
For this line, `usan` doesn't report about pointer arithmetic for NULL.

It indicates [a pointer index expression overflow](https://github.com/t-mat/lz4/runs/2683429036?check_suite_focus=true#step:3:106).

```
../lib/lz4.c:890:51: runtime error: pointer index expression with base 0x000001b7e530 overflowed to 0xffffffffffffe530
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior ../lib/lz4.c:890:51 in 
```

For instance, usan reports this error with the following variables:

```
dictionary=0x1d0e480, dictSize=65536, startIndex=30539776

  dictionary + dictSize - startIndex
= 0x1d0e480 + 65536 - 30539776 
= -7040
```